include/envoy/http/request_id_extension.h

```diff
@@ -5,19 +5,37 @@
 
 #include "envoy/common/pure.h"
 #include "envoy/http/header_map.h"
+#include "envoy/tracing/trace_reason.h"
 
 namespace Envoy {
 namespace Http {
 
-enum class TraceStatus { NoTrace, Sampled, Client, Forced };
+/**
+ * Request ID functionality available via stream info.
+ */
+class RequestIdStreamInfoProvider {
+public:
+  virtual ~RequestIdStreamInfoProvider() = default;
+
+  /**
+   * Perform a mod operation across the request id within a request and store the result in the
+   * provided output variable. This is used to perform sampling and validate the request ID.
+   * @param request_headers supplies the incoming request headers for retrieving the request ID.
+   * @param out reference to a variable where we store the result of the mod operation.
+   * @param mod integer to mod the request ID by.
+   * @return true if request ID is valid and out is populated by the result.
+   */
+  virtual bool modBy(const Http::RequestHeaderMap& request_headers, uint64_t& out,
```

Comment:
I don't have the full history here, but it seems that if we're going to make an extension, do we have any value in having every extension do the `%` operation? Should this method really be something like `obtainIntegerRepresentingRequestIdInHeadersToDoSomeModLater`? (Obviously shorter).

Reply:
That would work, though the vast majority of the complexity is still in "obtaining the integer" logic

Reply:
Agree and will change.